protos/orchestrator_service.proto

```diff
@@ -289,6 +289,31 @@ message TerminateResponse {
     // No payload
 }
 
+message QueryInstancesRequest {
+    InstanceQuery query = 1;
+    CancellationToken CancellationToken = 2;
+}
+
+message InstanceQuery{
+    repeated OrchestrationStatus runtimeStatus = 1;
+    google.protobuf.Timestamp createdTimeFrom = 2;
+    google.protobuf.Timestamp createdTimeTo = 3;
+    repeated google.protobuf.StringValue taskHubNames = 4;
+    int32 pageSize = 5;
+    google.protobuf.StringValue continuationToken = 6;
+    google.protobuf.StringValue instanceIdPrefix = 7;
+    bool fetchInputsAndOutputs = 8;
+}
+
+message CancellationToken{
+    //TODO: how should this message be defined
```

Comment:
Cancellation does not need to be modeled in the gRPC contract. It's implicit for all gRPC service methods. We can remove this.